### ISSUE TITLE
Not working with Nextclouds Tasks

### ISSUE THREAD

**Issue by LukeRuozzi** [melianmiko/ZeppOS-Tasks#41]
I successfully configured the connection with Nextcloud Tasks and applied the change to nginx reverse proxy configuration as reported in the Wiki but on my Amazfit Balance I got an error when opening the app:
"Cant' list calendars/TODO-lists, 412"

**Comment by programy-p** [comment]
Same here, I applied recomended config and got same error "Cant' list calendars/TODO-lists, 412" or ""Cant' list calendars/TODO-lists, 500". I use Nextcloud in Docker by Linuxserver.
using ChatGPT (i am to lame to understend nginx config by myself) I found out that nginx do not know and cant use "proxy_method".
Futhermore AI sugests that default nginx config got everything to "use" PUT, PROPFIND requests with CalDAV on Nextcloud as it simply redirects them to /remote.php in Nextcloud and Nextcloud can use them by default.
To put it simply by Nextcloud do no need any changes to ansewr calDAV reqiests like from ZeppOS Tasks.
So problem is either in Your app or I use wrong tasks app on Nextcloud server (but I use default tasks app from nextcloud shop).
Any odeas how I may debug problem more deeply to help make Your App work with Nextcloud?

**Comment by melianmiko** [comment]
> To put it simply by Nextcloud do no need any changes to ansewr calDAV reqiests like from ZeppOS Tasks.

ZeppOS doesn't support sending requests with PUT/DELETE/PROPFIND and other non-common HTTP methods, that's why it requires additional configuration at server.

I don't know why it stopped working, maybe guys from Nextcloud changed something, but I stopped using it and don't have time for researches.

**Comment by programy-p** [comment]
ok, thans for quick response :) 
I will try to investigte it with my little knowlage if I will find something usefull I will give some info.
nextcloud is fine but pain in the a.. when they upgrade something :/

**Comment by Pachinko0** [comment]
> ok, thans for quick response :) I will try to investigte it with my little knowlage if I will find something usefull I will give some info. nextcloud is fine but pain in the a.. when they upgrade something :/

Did you manage to find a solution?
I get the same errors here with Bip 6

**Comment by programy-p** [comment]
> > ok, thans for quick response :) I will try to investigte it with my little knowlage if I will find something usefull I will give some info. nextcloud is fine but pain in the a.. when they upgrade something :/
> 
> Did you manage to find a solution? I get the same errors here with Bip 6

I short NO :(

I am certain not to mess with Nextxloud server settings (I moved to AiO version it uses internal ngingx proxy to comunicate beteween their docker containers) it just blows everything up, usually beyond repair so do not try that aproach.

I tried to reconfig recomended proxy server Caddy on my OpnSense firewall router, but is uses caddy 2.0 that do not need any proxy settings - it just works as developer states and at the same time it just can't translate one request to anather, need plugins or somthing that do not exist or work in OpnSense enviroment. So it is a dead end now.

Now I am trying to build virtual nginx proxy server between opnsense (with caddy 2.0) and Nextcloud AiO (with internal nginx), like man in the middle, so it will get traffic routed from internet by firewall/ proxy, translate queries as melianmikon suggest, and then route it to Nextcloud to handle calDAV request. The problem is if my virtual proxy gets trafic it brakes connection with Nextcloud, so I still do not get tasks on watch but also cant connect to Nextcloud:)
But it is last hope for me, as I wont move from Nextcloud to google.
If I find a sollution I will post it here.

**Comment by Pachinko0** [comment]
> > > ok, thans for quick response :) I will try to investigte it with my little knowlage if I will find something usefull I will give some info. nextcloud is fine but pain in the a.. when they upgrade something :/
> > 
> > 
> > Did you manage to find a solution? I get the same errors here with Bip 6
> 
> I short NO :(
> 
> I am certain not to mess with Nextxloud server settings (I moved to AiO version it uses internal ngingx proxy to comunicate beteween their docker containers) it just blows everything up, usually beyond repair so do not try that aproach.
> 
> I tried to reconfig recomended proxy server Caddy on my OpnSense firewall router, but is uses caddy 2.0 that do not need any proxy settings - it just works as developer states and at the same time it just can't translate one request to anather, need plugins or somthing that do not exist or work in OpnSense enviroment. So it is a dead end now.
> 
> Now I am trying to build virtual nginx proxy server between opnsense (with caddy 2.0) and Nextcloud AiO (with internal nginx), like man in the middle, so it will get traffic routed from internet by firewall/ proxy, translate queries as melianmikon suggest, and then route it to Nextcloud to handle calDAV request. The problem is if my virtual proxy gets trafic it brakes connection with Nextcloud, so I still do not get tasks on watch but also cant connect to Nextcloud:) But it is last hope for me, as I wont move from Nextcloud to google. If I find a sollution I will post it here.

I was able to connect to nextcloud, but it retrieves blank data. I used the recommended Advanced nginx configuration.
Its sad because its my most wanted feature, i might return the watch for a wearos unfortunately

**Comment by stisa** [comment]
Same, i can connect to my task account but the Watch just shows an empty list page with 'Task list: ' written on top
